lib/pharos/upgrade_checker.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'open-uri'
+
+module Pharos
+  class UpgradeChecker
+    include Pharos::Logging
+
+    VERSION_URL = ENV["PHAROS_UPGRADE_CHECK_URL"] || 'https://get.pharos.sh/versions/latest'
+
+    def self.run
+      new(ENV['PHAROS_CHANNEL']).run
+    end
+
+    def initialize(channel = nil)
+      @uri = URI(VERSION_URL)
+      @uri.query = 'pre=true' if channel && channel.to_s == 'pre'
+    end
+
+    def run
+      logger.debug 'Checking for an upgrade ..'
+      latest_version = @uri.open('User-Agent' => "pharos/#{Pharos::VERSION}").read.chomp
```

Comment:
Would be more consistent with excon yes

Reply:
I've been thinking that it might be a good idea to rewrite `K8s::Transport` to get rid of excon.

As Mike Perham [said](https://www.mikeperham.com/2016/02/09/kill-your-dependencies/)

> Every Rails app pulls in a half dozen different HTTP clients: faraday, rest-client, httparty, excon, typhoeus, curb, etc. This is because various gems use them internally. A Rubygem should never use anything but Net::HTTP internally! Learn the Net:HTTP API, kill those dependencies and stop forcing extra HTTP client gems on your users.

Reply:
If we want people to use `k8s-client` for stuff.

Reply:
With a gem it might make sense but here, not so much (just a note that replacing Excon in k8s-client is not a simple task).